### ISSUE TITLE
Add Hugo import from wordpress

### ISSUE THREAD

**Issue by spf13** [gohugoio/hugo#100]
'hugo import ....'

**Comment by anthonyfok** [comment]
There is also a relatively new [WordPress to Hugo Exporter][], which works both as a WordPress plugin as as a PHP CLI script, which Cyrill Schumacher (@SchumacherFM) wrote based on the [WordPress to Jekyll Exporter](https://github.com/benbalter/wordpress-to-jekyll-exporter) that @xaprb has used about a year ago:

 * https://github.com/SchumacherFM/wordpress-to-hugo-exporter

[WordPress to Hugo Exporter]: https://github.com/SchumacherFM/wordpress-to-hugo-exporter

It must be pretty good: A recent [blog post](http://baty.net/2014/12/2014-12-31-my-hugo-experiment/) by Jack Baty (@jackbaty) has nothing but praises for @SchumacherFM's great tool:

> Converting from Wordpress was surprisingly easy. I used a plugin by Cyrill Schumacher and had all of my 1800+ posts, images, etc. converted for use with Hugo in less than an hour. All links were preserved so I don’t need to create a bunch of rewrite rules in Apache like every other time I’ve done this. Every step was easier than expected so I just kept going!

Moving forward, should we:

1. considered this issue solved, maybe except documenting the existence of this wonderful WordPress plugin/CLI tool?  Or...
2. create a `hugo import ...` command that would fetch the [WordPress to Hugo Exporter][] and guide the user to having it installed and run?  Or, even more ambitiously...
3. create a `hugo import ...` command which essentially does similar things as the PHP code, but all written in Go?

**Comment by battlemidget** [comment]
@anthonyfok i like 1 for the path of least resistance. most people are probably used to checking the website for importers similar to what jekyll displays on their site.

**Comment by anthonyfok** [comment]
Oh, one more thing: Jekyll *does* have its own WordPress importer:

 * http://import.jekyllrb.com/docs/wordpress/

as featured in http://www.smashingmagazine.com/2014/08/01/build-blog-jekyll-github-pages/

Though even in Jekyll's case, it seems that using the excellent [WordPress to Jekyll Exporter](https://github.com/benbalter/wordpress-to-jekyll-exporter) by GitHub staff Ben Balter (@benbalter) **is** the recommended route.

@battlemidget: Good idea!  Least amount of work for us, maybe perhaps writing a small tutorial.  No need to reinvent the wheel, especially if [WordPress to Hugo Exporter](https://github.com/SchumacherFM/wordpress-to-hugo-exporter) is already doing such a great job.

That said, if anyone really has the free time and *really* wants to do implement option 2 or option 3 as a Go programming exercise, I suppose that is welcome too.  :-)  But perhaps that time and energy is better spent on #101, a **Jekyll-to-Hugo importer**, as that no one has written yet.

Indeed, seeing that [Jekyll Import](http://import.jekyllrb.com/) already has the ability to convert from 23 other systems, we can leverage Jekyll Import to our advantage by perfecting our own Jekyll-to-Hugo Importer.  Once that is done, we'll have the ability to convert from 24 systems, all for the price of 1!  Haha!  :-)

**Comment by bjornerik** [comment]
I couldn't get the "Wordpress to Hugo Exporter" to work. Issues with newer WP versions.

**Comment by SchumacherFM** [comment]
I also don't use WordPress anymore ;-) and do not have the time to do further development on the converter script. If someone would like to have contributor permissions. Let me know.

**Comment by cgsmith** [comment]
hey @SchumacherFM I'd be interested in contributing and getting it be similar to ExitWP.   It would be nice to have it just parse the Wordpress XML or connect up from the local machine.

**Comment by SchumacherFM** [comment]
Cool @cgsmith :+1: 
Send me first a PR if all looks fine and can be merged I may add you as a collaborator.

**Comment by anthonyfok** [comment]
Status update:

@digitalcraftsman has kindly compiled a list of Hugo migration tools, and documented them at http://gohugo.io/tools/. See his Pull Request #1423 for more information.

**Comment by michaelsync** [comment]
I got timeout error when I use hugo exporter. https://github.com/SchumacherFM/wordpress-to-hugo-exporter/issues/8  ...

**Comment by Skarlso** [comment]
Hugo Exporter worked like a charm for me! However, there is a caveat. 

Your URLs must be in the format 2016/01/21/name-of-your-post. This will help massively. 

If the format is something like => ?p=123 it won't work well with Hugo.

**Comment by bep** [comment]
This issue has been automatically marked as stale because it has not been commented on for at least four months.

The resources of the Hugo team are limited, and so we are asking for your help.

If this is a **bug** and you can still reproduce this error on the <code>master</code> branch, please reply with all of the information you have about it in order to keep the issue open.

If this is a **feature request**, and you feel that it is still valuable, please open a proposal at https://discuss.gohugo.io/.

This issue will automatically be closed in four months if no further activity occurs. Thank you for all your contributions.

**Comment by bep** [comment]
**Note/Update:** This issue is marked as stale, and I may have said something earlier about "opening a thread on the discussion forum". Please don't.

If this is a **bug** and you can still reproduce this error on the latest <code>release</code> or the <code>master</code> branch, please reply with all of the information you have about it in order to keep the issue open.

If this is a **feature request**, and you feel that it is still relevant and valuable, please tell us why.

**Comment by eklausme** [comment]
I have written a converter program in the Go programming language to migrate from WordPress to Hugo. This is described here: https://eklausmeier.wordpress.com/2017/04/24/converting-wordpress-export-file-to-hugo/

I am very much interested in seeing this as `hugo import wordpress`.
